src/librustc_mir/interpret/traits.rs

```diff
@@ -61,16 +61,16 @@ impl<'a, 'mir, 'tcx, M: Machine<'a, 'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M>
         let drop = self.memory.create_fn_alloc(drop).with_default_tag();
         self.memory
             .get_mut(vtable.alloc_id)?
-            .write_ptr_sized(tcx, vtable, ptr_align, Scalar::Ptr(drop).into())?;
+            .write_ptr_sized(tcx, vtable, Scalar::Ptr(drop).into())?;
```

Comment:
Maybe add a comment that we do not bother with alignment checks here because anyway this is just generating static data.

Actually, with `Allocation` having so many methods, couldn't we do more of these preparations (writing all the stuff) *before* adding the allocation to the memory? We might even not add it to the memory at all but intern it immediately? But that's stuff for a future PR.

Reply:
Seems doable and probably even allows us to generalize this function so it can be used from codegen, too (without needing an eval context)